File: src/chapters/nonlinear-equations/content/code/fixed-point.go
package main

import (
	"fmt"
	"math"
)

// Fixed-point iteration x_{k+1} = g(x_k).
func fixedPoint(g func(float64) float64, x0, tol float64, maxIter int) float64 {
	x := x0
	for k := 0; k < maxIter; k++ {
		xn := g(x)
		if math.Abs(xn-x) < tol {
			return xn
		}
		x = xn
	}
	return x
}

func main() {
	fmt.Println(fixedPoint(math.Cos, 1.0, 1e-12, 200)) // Dottie number ~0.739085
}

File: src/chapters/nonlinear-equations/content/code/fixed-point.jl
function fixed_point(g, x0; tol = 1e-12, max_iter = 200)
    x = x0
    for k in 1:max_iter
        xn = g(x)
        abs(xn - x) < tol && return xn, k
        x = xn
    end
    return x, max_iter
end
println(fixed_point(cos, 1.0))   # Dottie number ~0.739085
